### PR TITLE
Added test checking if task is deleted, capsule sync status remains

### PR DESCRIPTION
**Pull request by sambible**
Automation for a recent RFE, https://bugzilla.redhat.com/show_bug.cgi?id=1956985
There is already an existing test, doing a similar check for repositories (from @swadeley ), which this test leverages. 

Bit of an ugly one, but setting up a capsule sync isn't a lightweight process, so this is the best I could see.

### CODE REVIEW

File: tests/foreman/api/test_contentmanagement.py [hunk 1]
```diff
@@ -1506,3 +1506,72 @@ def test_positive_sync_CV_to_multiple_LCEs(
...
+
+        :steps:
+            1. Create and Sync a Repo
+            2. Associate lifecycle env with capsule
+            3. Create a CV with the repo
+            4. Publish CV
+            5. Promote to lifecycle env
+            6. Sync Capsule
+            7. Delete the task using foreman-rake console
+            8. Verify the status of capsule is still synced
```
Comment (shubhamsg199) [inline tests/foreman/api/test_contentmanagement.py:1528]: We can add BZ here.
```suggestion
            8. Verify the status of capsule is still synced
            
        :BZ: 1956985
```

File: tests/foreman/api/test_contentmanagement.py [hunk 2]
```diff
@@ -1506,3 +1506,72 @@ def test_positive_sync_CV_to_multiple_LCEs(
...
+        """
+        repo_url = settings.repos.yum_1.url
+        repo = entities.Repository(product=function_product, url=repo_url).create()
```
Comment (shubhamsg199) [inline tests/foreman/api/test_contentmanagement.py:1531]: Switch to using sat object
```suggestion
        repo = target_sat.api.Repository(product=function_product, url=repo_url).create()
```

File: tests/foreman/api/test_contentmanagement.py [hunk 3]
```diff
@@ -1506,3 +1506,72 @@ def test_positive_sync_CV_to_multiple_LCEs(
...
+        module_capsule_configured.nailgun_capsule.content_add_lifecycle_environment(
+            data={'environment_id': function_lce.id}
+        )
+        result = module_capsule_configured.nailgun_capsule.content_lifecycle_environments()
+
+        assert len(result['results'])
+        assert function_lce.id in [capsule_lce['id'] for capsule_lce in result['results']]
+
+        cv = entities.ContentView(organization=function_org, repository=[repo]).create()
```
Comment (shubhamsg199) [inline tests/foreman/api/test_contentmanagement.py:1540]:
```suggestion
        cv = target_sat.api.ContentView(organization=function_org, repository=[repo]).create()
```

File: tests/foreman/api/test_contentmanagement.py [hunk 4]
```diff
@@ -1506,3 +1506,72 @@ def test_positive_sync_CV_to_multiple_LCEs(
...
+            f"""echo "ForemanTasks::Task.find('{search_result[0].id}')" | foreman-rake console"""
+        )
+        assert task_result.status == 0
+        assert 'RecordNotFound' in task_result.stdout
+
+        # Check sync status again, and ensure last_sync_time is still correct
+        sync_status = module_capsule_configured.nailgun_capsule.content_get_sync()
+        print(sync_status['last_sync_time'])
+        print('\r\n')
+        print(timestamp)
```
Comment (shubhamsg199) [inline tests/foreman/api/test_contentmanagement.py:1576]:
```suggestion
```

**Review by shubhamsg199 — CHANGES_REQUESTED** [review]

**Comment by vsedmik** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_contentmanagement.py::TestCapsuleContentManagement::test_positive_capsule_sync_status_persists

File: tests/foreman/api/test_contentmanagement.py (hunk 4 above)
Comment (lhellebr) [reply tests/foreman/api/test_contentmanagement.py:1576]: Or use logger if you want to have this logged.

Comment (sambible) [reply tests/foreman/api/test_contentmanagement.py:1576]: This is 100% some local testing stuff that got merged up :)

**Review by JacobCallahan — APPROVED** [review]
ACK. Thanks for coming back to this

**Review by omkarkhatavkar — APPROVED** [review]

**Pull request merged**